### PR TITLE
Enforce authenticated access

### PR DESCRIPTION
**Pull request by awead**
## Description

Defines access levels, previously known as _visibility_, on collections, and creates a central vocabulary and transaction for assigning them. Using Blacklight's access controls gem, these levels are used to restrict access to collections, works, and file sets based on whether or not a user is authenticated to Penn State or not.

Connected to #974 

## Changes

* rename visibility to access level
* create a RDF and repository vocabularies
* insert new operation to apply access levels to collections
* removes dependency on Valkyrie's access controls
* installs Blacklight's access controls
* enforces search, view, and edit restrictions on resources

### CODE REVIEW

File: app/cho/transaction/operations/shared/apply_access_level.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  module Operations
+    module Shared
+      class ApplyAccessLevel
+        include Dry::Transaction::Operation
+
+        def call(change_set)
+          return Success(change_set) if change_set.try(:access_level).blank?
+          return Failure(change_set) unless change_set.valid?
+
+          change_set.resource.read_groups += [change_set.access_level]
```
Comment (rschenk) [inline app/cho/transaction/operations/shared/apply_access_level.rb]: Is there a need to call `uniq` on this list? And/or is it a Set already? And/or does it matter if there are duplicates in this list?

File: app/cho/transaction/operations/shared/apply_access_level.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          Success(change_set)
+        rescue StandardError => e
+          change_set.errors.add(:apply_access_level, e.message)
+          Failure(change_set)
```
Comment (rschenk) [inline app/cho/transaction/operations/shared/apply_access_level.rb]: Helpful to use `Rejection` here?

File: app/devise/user.rb [hunk 3]
```diff
@@ -2,13 +2,14 @@
...
   include Blacklight::User
+  include Blacklight::AccessControls::User
   devise :http_header_authenticatable
 
   def groups
     return group_list.split(';?;') unless groups_last_update.blank? || ((Time.now - groups_last_update) > 24 * 60 * 60)
 
     populate_attributes
-    group_list.split(';?;')
+    group_list.split(';?;') + default_groups
```
Comment (rschenk) [inline app/devise/user.rb]: I'm not quite sure I follow what's going on here, but I just want to make sure that the `+ default_groups` bit does not also need to be in the guard clause up in line-9

File: app/cho/transaction/operations/shared/apply_access_level.rb (hunk 1 above)
Comment (awead) [reply app/cho/transaction/operations/shared/apply_access_level.rb]: Great catch! In short, no. A `Set` already enforces uniqness, and it probably wouldn't matter anyway if there were dups, but it's nice to know that there won't be.

File: app/cho/transaction/operations/shared/apply_access_level.rb (hunk 2 above)
Comment (awead) [reply app/cho/transaction/operations/shared/apply_access_level.rb]: Yep, I think so. I'll change that...

File: app/devise/user.rb (hunk 3 above)
Comment (awead) [reply app/devise/user.rb]: `group_list` is a method from the PsuDir gem, and is completely *not* obvious at all here. It's a listing of groups from our LDAP server. We split those and add the default groups to that. I think some comment might be helpful here.

**Pull request merged**